### PR TITLE
Added folder for handling outliers

### PR DESCRIPTION
**Pull request by Priyesh1202**

### CODE REVIEW

File: preprocessy/outliers/handleoutlier.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+class HandleOutlier:
+
+
+    """ Class for handling outliers on its own or according to users needs.
+
+        Private methods
+       _ _ _ _ _ _ _ _ _ _
+
+       __returnQuartiles() : returns the 5% and 95% mark in the distrubution
```
Comment (Saif807380) [inline preprocessy/outliers/handleoutlier.py:9]: No such method exists in this class. The method name is `__return_quartiles`.

File: preprocessy/outliers/handleoutlier.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        Private methods
+       _ _ _ _ _ _ _ _ _ _
+
+       __returnQuartiles() : returns the 5% and 95% mark in the distrubution 
+                             of data(Values above are default values)
+
+        Public Methods
+       _ _ _ _ _ _ _ _ _
+
+       handleOutliers() : Takes in the dataset as input, finds the quartiles
```
Comment (Saif807380) [inline preprocessy/outliers/handleoutlier.py:15]: No such method exists in this class. The method name is `handle_quartiles`.

File: preprocessy/outliers/handleoutlier.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                -replace : default = False => if user wants to replace 
+                           outliers with -999 everywhere instead of 
+                           removing them
+                -q1 : specify the starting range (Default is 0.05)
+                -q3 : specify the end of the range (Default is 0.95)
+
+        """
+        self.train_df = train_df
+        self.cols = []
+        self.removeoutliers = True
```
Comment (Saif807380) [inline preprocessy/outliers/handleoutlier.py:45]: We have used the `snake case` for variable names as well.

File: preprocessy/outliers/handleoutlier.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        self.replace = False
+        self.quartiles = {}
+        self.qone = 0.05
+        self.qthree = 0.95
```
Comment (Saif807380) [inline preprocessy/outliers/handleoutlier.py:49]: Variable could renamed to `third_quartile`.

File: preprocessy/outliers/handleoutlier.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                -q1 : specify the starting range (Default is 0.05)
+                -q3 : specify the end of the range (Default is 0.95)
+
+        """
+        self.train_df = train_df
+        self.cols = []
+        self.removeoutliers = True
+        self.replace = False
+        self.quartiles = {}
+        self.qone = 0.05
```
Comment (Saif807380) [inline preprocessy/outliers/handleoutlier.py:48]: Variable could be renamed to `first_quartile`.

File: preprocessy/outliers/handleoutlier.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        self.qthree = 0.95
+        if 'cols' in params.keys():
+            self.cols = params['cols']
+        if 'removeoutliers' in params.keys():
+            self.removeoutliers = params['removeoutliers']
```
Comment (Saif807380) [inline preprocessy/outliers/handleoutlier.py:53]: Fix case of the variable.

File: preprocessy/outliers/handleoutlier.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+    def __return_quartiles(self,col):
+        # return the quartile range or q1 and q3 values for the column passed as parameter
+        quartiles = self.quartiles
+        train_df = self.train_df
+        q1 = train_df[col].quantile(self.qone)
+        q1 = round(q1)
+        q3 = train_df[col].quantile(self.qthree)
+        q3 = round(q3)
+        quartiles[col] = [q1,q3]
```
Comment (Saif807380) [inline preprocessy/outliers/handleoutlier.py:69]: `quartiles` is a local variable whose scope does not exist beyond `__return_quartiles`.  Initially `self.quartiles` is empty. In every call to `__return_quartiles`, `quartiles` is initialized as an empty dictionary. `self.quartiles` is never updated.

File: preprocessy/outliers/handleoutlier.py [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        return True
```
Comment (Saif807380) [inline preprocessy/outliers/handleoutlier.py:70]: You can simply remove `True` if the method does not return anything.

File: preprocessy/outliers/handleoutlier.py [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    def handle_outliers(self):
+
+        ### parameters till now: train_df, cols, removeoutliers, replace
+        train_df = self.train_df
+        if self.removeoutliers: # if user has marked removeoutliers = True and wants outliers removed..
+            if len(self.cols)>=1:
+                for col in self.cols:
+                    self.__return_quartiles(col)
+                for col in self.cols:
+                    q = self.quartiles[col]
```
Comment (Saif807380) [inline preprocessy/outliers/handleoutlier.py:81]: This line will give a `keyError` as `self.quartiles` is empty.

File: preprocessy/outliers/handleoutlier.py [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+                    train_df = train_df[(train_df[col]<=q3)]
+
+        # if removeoutliers = False and replace=True ie user wants outliers replaced by a value to indicate these
+        # are outliers
+        elif self.replace: 
+            if len(self.cols)>=1:
+                for col in self.cols:
+                    self.__return_quartiles(col)
+                for col in self.cols:
+                    q = self.quartiles[col]
```
Comment (Saif807380) [inline preprocessy/outliers/handleoutlier.py:94]: This line will give a `keyError` as `self.quartiles` is empty.

**Review by Saif807380 — CHANGES_REQUESTED** [review]
Please update `__init__.py` file. Refer the `__init__.py` file of other modules.

File: preprocessy/outliers/test.py [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    'replace' :False,
+    'q1' : 0.10,
+    'q3' : 0.90
+}
+train_df = [1,2,3,100,0,3,6,4,20]
+train_df = DataFrame(data=train_df, columns=['Test'])
+k = HandleOutlier(train_df,params=params)
+train = k.handle_outliers()
+print(train_df)
+print(train)
```
Comment (Saif807380) [inline preprocessy/outliers/test.py:16]: Please add tests in the `tests` directory. Rename the file to `test_outliers.py`.

File: preprocessy/outliers/handleoutlier.py (hunk 1 above)
Comment (Priyesh1202) [reply preprocessy/outliers/handleoutlier.py:9]: Will correct

File: preprocessy/outliers/handleoutlier.py (hunk 2 above)
Comment (Priyesh1202) [reply preprocessy/outliers/handleoutlier.py:15]: Will correct

File: preprocessy/outliers/handleoutlier.py (hunk 3 above)
Comment (Priyesh1202) [reply preprocessy/outliers/handleoutlier.py:45]: Will correct

File: preprocessy/outliers/handleoutlier.py (hunk 4 above)
Comment (Priyesh1202) [reply preprocessy/outliers/handleoutlier.py:49]: Will correct

File: preprocessy/outliers/handleoutlier.py (hunk 5 above)
Comment (Priyesh1202) [reply preprocessy/outliers/handleoutlier.py:48]: Will correct

File: preprocessy/outliers/handleoutlier.py (hunk 6 above)
Comment (Priyesh1202) [reply preprocessy/outliers/handleoutlier.py:53]: Will correct

File: preprocessy/outliers/handleoutlier.py (hunk 7 above)
Comment (Priyesh1202) [reply preprocessy/outliers/handleoutlier.py:69]: Ahhh nice one will correct

File: preprocessy/outliers/handleoutlier.py (hunk 8 above)
Comment (Priyesh1202) [reply preprocessy/outliers/handleoutlier.py:70]: Will correct

File: preprocessy/outliers/handleoutlier.py (hunk 9 above)
Comment (Priyesh1202) [reply preprocessy/outliers/handleoutlier.py:81]: Related to the local scope if quartiles variables will be consequently corrected

File: preprocessy/outliers/handleoutlier.py (hunk 10 above)
Comment (Priyesh1202) [reply preprocessy/outliers/handleoutlier.py:94]: Will correct

File: preprocessy/outliers/test.py (hunk 11 above)
Comment (Priyesh1202) [reply preprocessy/outliers/test.py:16]: Yep

**Review by Saif807380 — APPROVED** [review]
Good Work!

**Pull request merged**